bin/reth/src/node/mod.rs

```diff
@@ -263,19 +269,27 @@ impl<Ext: RethCliExt> NodeCommand<Ext> {
         // setup the blockchain provider
         let factory = ProviderFactory::new(Arc::clone(&db), Arc::clone(&self.chain));
         let blockchain_db = BlockchainProvider::new(factory, blockchain_tree.clone())?;
-
         let blob_store = InMemoryBlobStore::default();
-        let transaction_pool = reth_transaction_pool::Pool::eth_pool(
-            TransactionValidationTaskExecutor::eth_with_additional_tasks(
-                blockchain_db.clone(),
-                Arc::clone(&self.chain),
-                blob_store.clone(),
-                ctx.task_executor.clone(),
-                1,
-            ),
-            blob_store,
-            self.txpool.pool_config(),
+
+        let eth_builder = TransactionValidationTaskExecutor::eth_builder(Arc::clone(&self.chain));
+        // Overwrites the trusted setup file.
+        let eth_builder = match self.trusted_setup_file {
+            Some(ref trusted_setup_file) => {
+                let kzg_settings = self.load_kzg_settings(trusted_setup_file)?;
+                info!(target: "reth::cli", "KZG trusted setup file loaded");
+
+                eth_builder.kzg_settings(kzg_settings)
+            }
+            None => eth_builder,
+        };
+        let validator = eth_builder.with_additional_tasks(1).build_with_tasks(
+            blockchain_db.clone(),
+            ctx.task_executor.clone(),
+            blob_store.clone(),
```

Comment:
@mattsse decided to go with this since I think it's simpler and clearer. Lmk if it's too verbose, and I'll extract it to a separate function.